lib/core/src/animation/animator/easing.rs

```diff
@@ -0,0 +1,88 @@
+//! This file provides the implementation of EasingAnimator.
+
+use super::ContinuousAnimator;
+use crate::animation::position::HasPosition;
+use crate::animation::easing::FnEasing;
+use crate::control::event_loop::EventLoop;
+use crate::math::utils::linear_interpolation;
+
+use nalgebra::Vector3;
+use nalgebra::clamp;
+use std::rc::Rc;
+use std::cell::RefCell;
+
+
+// ==========================
+// === EasingAnimatorData ===
+// ==========================
+
+struct EasingAnimatorData {
+    object              : Box<dyn HasPosition>,
```

Comment:
I don't like this. EasingAnimatorData should work on anything, not necessary something which has position. I want for example to connect it to a Div opacity value. It should accept a lambda which it should call with a new value instead.

Reply:
And why the object is boxed instead of being just generic parameter?

Reply:
@farmaazon it's just for the case we want to reuse EasingAnimator and change it with other object that also implement `HasPosition`. We can't do that when after monomorphization. But I gotta ask: What's the trade-off between using boxed dyn and using generic parameters?

About the flexibility, I will replace `HasPosition` with a new trait called `Interpolatable` and extend our example to show it animating different properties.